src/Framework/AspNetCoreAnalyzers/src/CodeFixes/Dependencies/AddPackageFixer.cs

```diff
@@ -0,0 +1,113 @@
+    {
+        var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+        if (root == null)
+        {
+            return;
+        }
+
+        var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+        if (semanticModel == null)
+        {
+            return;
+        }
+
+        var wellKnownTypes = WellKnownTypes.GetOrCreate(semanticModel.Compilation);
+        var wellKnownExtensionMethodCache = ExtensionMethodsClass.ConstructFromWellKnownTypes(wellKnownTypes);
+
+        foreach (var diagnostic in context.Diagnostics)
+        {
+            var location = diagnostic.Location.SourceSpan;
+            var node = root.FindNode(location);
+            if (node == null)
+            {
+                return;
+            }
+            var methodName = node is IdentifierNameSyntax identifier ? identifier.Identifier.Text : null;
+            if (methodName == null)
+            {
+                return;
+            }
+
+            if (node.Parent is not MemberAccessExpressionSyntax)
+            {
+                return;
+            }
+
+            var symbol = semanticModel.GetSymbolInfo(((MemberAccessExpressionSyntax)node.Parent).Expression).Symbol;
+            var symbolType = symbol switch
+            {
+                IMethodSymbol methodSymbol => methodSymbol.ReturnType,
+                IPropertySymbol propertySymbol => propertySymbol.Type,
+                ILocalSymbol localSymbol => localSymbol.Type,
+                _ => null
+            };
+
+            if (symbolType == null)
+            {
+                return;
+            }
+
+            var targetThisAndExtensionMethod = new ThisAndExtensionMethod(symbolType, methodName);
+            if (wellKnownExtensionMethodCache.TryGetValue(targetThisAndExtensionMethod, out var packageSourceAndNamespace))
+            {
+                var position = diagnostic.Location.SourceSpan.Start;
+                var packageInstallData = new AspNetCoreInstallPackageData(
+                    packageSource: null,
+                    packageName: packageSourceAndNamespace.packageName,
+                    packageVersionOpt: null,
+                    packageNamespaceName: packageSourceAndNamespace.namespaceName);
+                var codeAction = await TryCreateCodeActionAsync(
+                    context.Document,
+                    position,
+                    packageInstallData,
+                    context.CancellationToken);
+
+                if (codeAction != null)
+                {
+                    context.RegisterCodeFix(codeAction, diagnostic);
+                }
+            }
+
+        }
+    }
+
+    public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+    public override ImmutableArray<string> FixableDiagnosticIds { get; } = ["CS1061"];
```

Comment:
nit: A comment of what this diagnostic ID is

Reply:
Good tip -- Added an example of the emitted diagnostic we target here.